File: APIs/models/dtos/item.py
from database import db


class Item(db.Model):
    __tablename__ = "item"
    item_code = db.Column(db.String(100), primary_key=True)
    item_name = db.Column(db.String(100))
    item_type = db.Column(db.String(100))
    series = db.Column(db.String(100))
    price = db.Column(db.Numeric(10, 0))
    jan_code = db.Column(db.String(100))

    def to_dict(self):
        return {
            "item_code": self.item_code,
            "item_name": self.item_name,
            "item_type": self.item_type,
            "series": self.series,
            "price": self.price,
            "jan_code": self.jan_code,
        }

File: front_app/lib/models/item.dart
import 'package:decimal/decimal.dart';
import 'package:flutter_application_1/models/table_item.dart';

import '../constants.dart';

class Item extends TableItem {
  final String itemCode;
  final String itemName;
  final String type;
  final String series;
  final Decimal price;
  final String janCode;
  final String image;

  Item(
      {required this.itemCode,
      required this.itemName,
      required this.type,
      required this.series,
      required this.price,
      required this.janCode,
      required this.image});

  factory Item.fromJson(Map<String, dynamic> json) {
    return Item(
        itemCode: json['item_code'],
        itemName: json['item_name'],
        type: json['item_type'],
        series: json['series'],
        price: Decimal.parse(json['price'] ?? '0'),
        janCode: json['jan_code'],
        image: "${API.ITEM_IMAGE}${json['item_code']}.jpg");
  }
  @override
  Map<String, dynamic> toTableData() {
    return {
      'itemCode': itemCode,
      'itemName': itemName,
      'type': type,
      'series': series,
      'price': price,
      'janCode': janCode,
      'image': image
    };
  }

  static const columnsForLargeScreen = [
    'itemCode',
    'itemName',
    'type',
    'series',
    'price',
    'janCode',
    'image',
  ];

  static const columnsForMediumScreen = [
    'itemCode',
    'itemName',
    'type',
    'series',
    'price',
    'janCode',
    'image'
  ];

  static const columnsForSmallScreen = ['itemCode', 'itemName', 'image'];
}
